src/common/src/cert.rs

```diff
@@ -0,0 +1,39 @@
+use crate::constants::DEFAULT_CERT_DIR;
+use anyhow::Result;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+/* TODO: The private key at least needs to be stored in the cache and encrypted */
```

Comment:
This needs work. We should store the device private key somewhere more secure.

Reply:
Will need to work with Kanidm to add these to the storage. This MR is **blocked** until we have a proper way of storing the certificate and private key. This WIP request is storing the private key readable by anyone. Obviously this is wrong. Will need to send the key to the daemon and have it stored securely, with the daemon only having read permissions.

Reply:
Need to use [kanidm/hsm-crypto](https://github.com/kanidm/hsm-crypto).